### PR TITLE
Spring boot Security HW

### PR DESCRIPTION
**Pull request by araslav**

### CODE REVIEW

File: src/main/java/book/store/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -3,21 +3,32 @@
...
+    private final RoleRepository roleRepository;
 
     @Override
     public UserResponseDto save(UserRegistrationRequestDto userDto) {
-        User user = userRepository.save(userMapper.toModel(userDto));
+        User user = userMapper.toModel(userDto);
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+        if (user.getRoles() == null || user.getRoles().isEmpty()) {
+            user.setRoles(Set.of(roleRepository.findByName(Role.RoleName.USER)));
+        }
```
Comment (Rommelua) [inline src/main/java/book/store/service/impl/UserServiceImpl.java]:
```suggestion
            user.setRoles(Set.of(roleRepository.findByName(Role.RoleName.USER)));   
```

File: src/main/java/book/store/model/User.java [hunk 2]
```diff
@@ -2,18 +2,27 @@
...
 import lombok.Setter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
 @Getter
 @Setter
 @Entity
 @Table(name = "users")
-public class User {
+public class User implements UserDetails {
```
Comment (Rommelua) [inline src/main/java/book/store/model/User.java:29]: implement soft delete on class level

File: src/main/java/book/store/config/SecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    public PasswordEncoder getPasswordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        return http
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+
```
Comment (koval-kn) [inline src/main/java/book/store/config/SecurityConfig.java:32]:
```suggestion
```

**Review by koval-kn — APPROVED** [review]

**Comment by araslav** [comment]
Done

**Pull request merged**